### ISSUE TITLE
Uncorrect format  in energy dependent `LightCurveTemplateTemporalModel`

### ISSUE THREAD

**Issue by fabiopintore** [gammapy/gammapy#5053]
The `LightCurveTemplateTemporalModel.to_dict` allows to chose the format (i.e. `table` or `map`, with the former as default) of the model. 
However, the `SkyModel.to_dict()`, which inherits `LightCurveTemplateTemporalModel.to_dict`, only allows for the default format (`table`). This is not suitable in case of energy-dependent `LightCurveTemplateTemporalModel`, where the `map` format should be selected.

A possibility could be to modify `temporal_model.to_dict` in order to remove the `format` argument and let the function to select the proper format.

To replicate the issue:

```
pointing_position = SkyCoord("100 deg", "30 deg", frame="icrs")
spatial_model = PointSpatialModel.from_position(pointing_position)
spectral_model = ConstantSpectralModel(const="1 cm-2 s-1 TeV-1")
temporal_model = LightCurveTemplateTemporalModel.read("$GAMMAPY_DATA/gravitational_waves/GW_example_DC_map_file.fits.gz", format="map")

model = SkyModel(
    spatial_model=spatial_model,
    spectral_model=spectral_model,
    temporal_model=temporal_model,
    name="test",
)

model.to_dict()
```

**Comment by AtreyeeS** [comment]
So the issue here is that the `SkyModel.to_dict` does not propagate options to the individual `to_dict` functions it calls?

For this specific case, I suppose it is reasonable to add a `LightCurveTemplateTemporalModel.guess_format` which return `map` if an energy column is present, and the other functions use the same if `format` is unspecified

**Comment by registerrier** [comment]
This is solved in #5062. Closing.
